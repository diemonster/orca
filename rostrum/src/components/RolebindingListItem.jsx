import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { rolebindingDelete } from '../actions/rolebindingActions';


export class RolebindingListItem extends React.Component {
  constructor(props) {
    super(props);

    this.handleDelete = this.handleDelete.bind(this);
  }

  handleDelete() {
    const {
      dispatchRolebindingDelete,
      rolebindingName,
      selectedNamespace,
    } = this.props;
    dispatchRolebindingDelete(selectedNamespace, rolebindingName);
  }

  render() {
    const {
      rolebindingName,
    } = this.props;

    return (
      <li className="rolebinding-list-item">
        {rolebindingName}
        <button
          type="button"
          className="rolebinding-delete-button"
          onClick={this.handleDelete}
        >
        X
        </button>
      </li>
    );
  }
}

RolebindingListItem.propTypes = {
  dispatchRolebindingDelete: PropTypes.func.isRequired,
  rolebindingName: PropTypes.string.isRequired,
  selectedNamespace: PropTypes.string.isRequired,
};

const mapStateToProps = state => ({
  selectedNamespace: state.namespace.selectedNamespace,
});

const mapDispatchToProps = dispatch => ({
  dispatchRolebindingDelete: (namespace, rolebinding) => {
    dispatch(rolebindingDelete(namespace, rolebinding));
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(RolebindingListItem);
