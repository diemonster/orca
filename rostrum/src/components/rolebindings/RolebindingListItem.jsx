import React from 'react';

function RolebindingListItem(props) {
  const {
    rolebinding,
  } = props;

  return (
    <li className={`rolebinding-list-item`}>
      {rolebinding}
    </li>
  );
}

export default RolebindingListItem;
