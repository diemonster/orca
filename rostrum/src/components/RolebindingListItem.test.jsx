import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { RolebindingListItem } from './RolebindingListItem';


Enzyme.configure({ adapter: new Adapter() });

describe('components', () => {
  describe('RolebindingListItem', () => {
    let props;

    beforeEach(() => {
      props = {
        dispatchRolebindingDelete: jest.fn(),
        rolebindingName: 'some-rolebinding',
        selectedNamespace: 'some-namespace',
      };
    });

    it('should render itself and its subcomponents', () => {
      const rolebindingListItem = shallow(<RolebindingListItem {...props} />);
      expect(rolebindingListItem).toMatchSnapshot();
    });

    it('should dispatch an action to delete a rolebinding on delete button click', () => {
      const mockFn = jest.fn();
      const event = {
        preventDefault: jest.fn(),
      };

      props.dispatchRolebindingDelete = mockFn;
      const rolebindingListItem = shallow(<RolebindingListItem {...props} />);

      rolebindingListItem.find('button').simulate('click', event);
      expect(mockFn).toHaveBeenCalledWith('some-namespace', 'some-rolebinding');
    });
  });
});
