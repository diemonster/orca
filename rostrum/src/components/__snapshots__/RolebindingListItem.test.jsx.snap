// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components RolebindingListItem should render itself and its subcomponents 1`] = `
<li
  className="rolebinding-list-item"
>
  <button
    className="rolebinding-delete-button"
    onClick={[Function]}
    type="button"
  >
    X
  </button>
  some-rolebinding
</li>
`;
